app/src/main/java/co/runloop/influencer/fragment/ContactsFragment.java

```diff
@@ -54,21 +58,37 @@ public View onCreateView(@NonNull LayoutInflater inflater,
         recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
         recyclerView.setAdapter(adapter);
 
+        progressBar = root.findViewById(R.id.frag_contacts_list_progressbar);
+
         contactsViewModel = ViewModelProviders
                 .of(this)
                 .get(ContactsViewModel.class);
-        contactsViewModel.getContacts().observe(this, (@Nullable List<Contact> contacts) -> {
-            adapter.setContacts(contacts);
-            adapter.notifyDataSetChanged();
-        });
-        if (contactsIsAvailable()) {
-            contactsViewModel.onPermissionGranted();
-            if(contactsViewModel.getContacts().getValue() == null) {
-                contactsViewModel.loadAll();
+        contactsViewModel.getContacts().observe(this, (@Nullable Resource<List<Contact>> contacts) -> {
+            if (contacts == null) {
+                progressBar.setVisibility(View.GONE);
```

Comment:
So, you're waiting here for the first async call to return before you show the progress indicator. It may be better UX if the progress indicator will start before calling `getContacts` (although I am not sure there will be a noticeable difference)

Reply:
More so, if the contacts list returns as null, does the loading process actually continue? It seems that it doesn't, in which case the progress indicator is "lying", and an error screen should be presented, no?

Reply:
1 - I'm not sure to start progress "by default", in theory here can be happens many other things before contacts start load, maybe even we need to press "Load contacts" for example

2 - Do you think if contacts returned as null, we should look on it as on error ?

Reply:
1. I understand what you mean.
2. Since we control the provider and the results it will return, I think that an empty list should represent "no contacts" (for whichever reason), and null should only happen if we have an error of some sort. Maybe we can prevent this list from ever being null?

Reply:
Inside provider can not be situation when it returns null, it returns empty list by default.
But livedata marked as @Nulllable, because null can be posted to observer. 
It looks next - data inside livedata cannot be null, but livedata can. 
I don't send nulls to observer, but in theory it can happens.
Should we check null then ?

Reply:
I guess it's better to add a redundant null check than to theoretically crash :)